File: test/react-native/features/fixtures/scenario-launcher/ios/Scenarios/NativePromiseRejectionHandledScenario.m
#import "NativePromiseRejectionHandledScenario.h"

@implementation NativePromiseRejectionHandledScenario

- (void)run: (RCTPromiseResolveBlock)resolve
     reject:(RCTPromiseRejectBlock)reject {
  reject(@"NativeError", @"NativePromiseRejectionHandledScenario", [NSError errorWithDomain:@"com.example" code:408 userInfo:nil]);
}

@end

File: test/react-native/features/fixtures/ios-module/Scenarios/NativePromiseRejectionHandledScenario.swift
//
//  NativePromiseRejectionHandledScenario.swift
//  reactnative
//
//  Created by Nick Dowell on 02/12/2020.
//

class NativePromiseRejectionHandledScenario: Scenario {
  
  override func run(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    reject("NativeError", "NativePromiseRejectionHandledScenario", NSError(domain:"com.example", code:408))
  }
}
